src/main/java/com/monumental/services/TagService.java

```diff
@@ -86,24 +45,24 @@ public TagService(SessionFactoryService sessionFactoryService) {
         return this.getWithCriteriaQuery(query, 10);
     }
 
-    @Transactional
+    /**
+     * Safely create a new tag without duplication
+     * @param name          The name of the tag to create
+     * @param monuments     The list of monuments to associate it with
+     * @param isMaterial    Whether or not the tag is a material
+     */
     public Tag createTag(String name, List<Monument> monuments, Boolean isMaterial) {
-        try {
-            List<Tag> duplicates = this.tagRepository.getAllByNameAndIsMaterial(name, isMaterial);
-            Tag tag;
-            if (duplicates != null && duplicates.size() > 0) {
-                tag = duplicates.get(0);
-            } else {
-                tag = new Tag();
-                tag.setName(name);
-                tag.setIsMaterial(isMaterial);
-            }
-            tag.getMonuments().addAll(monuments);
-            this.tagRepository.save(tag);
-            return tag;
-        } catch (Exception e) {
-            e.printStackTrace();
-            return null;
+        List<Tag> duplicates = this.tagRepository.getAllByNameAndIsMaterial(name, isMaterial);
+        Tag tag;
+        if (duplicates != null && duplicates.size() > 0) {
+            tag = duplicates.get(0);
+        } else {
+            tag = new Tag();
+            tag.setName(name);
+            tag.setIsMaterial(isMaterial);
         }
+        tag.getMonuments().addAll(monuments);
+        this.tagRepository.save(tag);
+        return tag;
     }
 }
```

Comment:
Same thing as with `MonumentService`, is there a need for Service classes and Repository classes now?

Reply:
Yes, the Services should provide any functionality not supported by the Repository since teh Repository is an interface (the actual implementation of the interface is done by JPA, which we can't add onto unfortunately)